purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesOrchestrator.kt

```diff
@@ -194,22 +197,24 @@ internal class PurchasesOrchestrator constructor(
             state = state.copy(appInBackground = false, firstTimeInForeground = false)
         }
         log(LogIntent.DEBUG, ConfigureStrings.APP_FOREGROUNDED)
-        if (shouldRefreshCustomerInfo(firstTimeInForeground)) {
-            log(LogIntent.DEBUG, CustomerInfoStrings.CUSTOMERINFO_STALE_UPDATING_FOREGROUND)
-            customerInfoHelper.retrieveCustomerInfo(
-                identityManager.currentAppUserID,
-                fetchPolicy = CacheFetchPolicy.FETCH_CURRENT,
-                appInBackground = false,
-                allowSharingPlayStoreAccount = allowSharingPlayStoreAccount,
-            )
-        }
-        offeringsManager.onAppForeground(identityManager.currentAppUserID)
-        postPendingTransactionsHelper.syncPendingPurchaseQueue(allowSharingPlayStoreAccount)
-        synchronizeSubscriberAttributesIfNeeded()
-        offlineEntitlementsManager.updateProductEntitlementMappingCacheIfStale()
-        flushPaywallEvents()
-        if (firstTimeInForeground && isAndroidNOrNewer()) {
-            diagnosticsSynchronizer?.syncDiagnosticsFileIfNeeded()
+        enqueue {
```

Comment:
I don't think there should be any issue with enqueuing all this right? It is all pretty much updating caches, which is already changing into another thread to make the API call. This way we guarantee the accesses to the deviceCache are in the background